tools/GenerateExternalContributors.ps1

```diff
@@ -0,0 +1,123 @@
+$rootPath = "$PSScriptRoot\.."
+$changeLogFile = Get-Item -Path "..\ChangeLog.md"
+$changeLogContent = Get-Content -Path $changeLogFile.FullName | Out-String
+
+Write-Host -ForegroundColor Green 'Create ExternalContributors.md'
+# Create md file to store contributors information.
+$contributorsMDFile = Join-Path $PSScriptRoot 'ExternalContributors.md'
+if ((Test-Path -Path $contributorsMDFile)) {
+    Remove-Item -Path $contributorsMDFile -Force
+}
+New-Item -ItemType "file" -Path $contributorsMDFile
+
+$commitsUrl = "https://api.github.com/repos/Azure/azure-powershell/commits?since=$SinceDateStr&sha=$Branch"
+$token = ConvertTo-SecureString $AccessToken -AsPlainText -Force
+# Get last page number of commints.
+$commintsPagesLink = (Invoke-WebRequest -Uri $commitsUrl -Authentication Bearer -Token $token).Headers.Link
+$commintsLastPageNumber = 1 # Default value
+if (![string]::IsNullOrEmpty($commintsPagesLink)) {
+    if ($commintsPagesLink.LastIndexOf('&page=') -gt 0) {
+        [int]$commintsLastPageNumber = $commintsPagesLink.Substring($commintsPagesLink.LastIndexOf('&page=') + '&page='.Length, 1) 
+    }
+}
+$PRs = @()
+for ($pageNumber=1; $pageNumber -le $commintsLastPageNumber; $pageNumber++) {
+    $commitsPageUrl = $commitsUrl + "&page=$pageNumber"
+    $PRs += Invoke-RestMethod -Uri $commitsPageUrl -Authentication Bearer -Token $token -ResponseHeadersVariable 'ResponseHeaders'
+}
+
+$sortPRs = $PRs | Sort-Object -Property @{Expression = {$_.author.login}; Descending = $False}
+
+$skipContributors = @('aladdindoc')
+
+$contributorsMDHeaderFlag = $True
+for ($PR = 0; $PR -lt $sortPRs.Length; $PR++) {
+    if ($skipContributors.Contains($sortPRs[$PR].author.login))
+    {
+        continue
+    }
+    Invoke-RestMethod -Uri "https://api.github.com/orgs/Azure/members/$($sortPRs[$PR].author.login)" -Authentication Bearer -Token $token -ResponseHeadersVariable 'ResponseHeaders' -StatusCodeVariable 'StatusCode' -SkipHttpErrorCheck > $null
+    if ($StatusCode -eq '204') {
+        # Add internal contributors to skipContributors to reduce the number of https requests sent.
+        $skipContributors += $sortPRs[$PR].author.login
+        continue
+    }
+    if ($contributorsMDHeaderFlag) {
+        Write-Host -ForegroundColor Green 'Output exteneral contributors infomation.'
+        '### Thanks to our community contributors' | Out-File -FilePath $contributorsMDFile -Force
+        Write-Host '### Thanks to our community contributors'
+        $contributorsMDHeaderFlag = $False
+    }
+    $account = $sortPRs[$PR].author.login
+    $name = $sortPRs[$PR].commit.author.name
+    $index = $sortPRs[$PR].commit.message.IndexOf("`n`n")
+    if ($index -lt 0) {
+        $commitMessage = $sortPRs[$PR].commit.message
+    } else {
+        $commitMessage = $sortPRs[$PR].commit.message.Substring(0, $index)
+    }
+    # Skip already exits commits.
+    if ($changeLogContent.Contains($commitMessage.Substring($commitMessage.LastIndexOf('('))))
```

Comment:
I didn't get the meaning of this block. 

Can  you get the PR id from commit instead of message? Sometimes PR message may have format like "... (#A) (#B)"
Where did you use `CommitSkip`?

Reply:
Modified code
https://github.com/Azure/azure-powershell/blob/a07a7a42cdeccec04334bbe74457d3af05d2a567/tools/GenerateExternalContributors.ps1#L54